.travis.yml

```diff
@@ -62,89 +86,128 @@ matrix:
       env:
         - TEST_SLOW="true"
         - SPLIT="3/3"
-    - python: 3.4
+    - python: 3.5
       env:
         - TEST_SLOW="true"
         - SPLIT="1/3"
-    - python: 3.4
+    - python: 3.5
       env:
         - TEST_SLOW="true"
         - SPLIT="2/3"
-    - python: 3.4
+    - python: 3.5
       env:
         - TEST_SLOW="true"
         - SPLIT="3/3"
+
     - python: 2.7
       env:
-        - TEST_THEANO="true"
-      addons:
-        apt:
-          packages:
-          - libatlas-dev
-          - libatlas-base-dev
-          - liblapack-dev
-          - gfortran
-    - python: 3.4
-      env:
-        - TEST_THEANO="true"
+        - TEST_AUTOWRAP="true"
       addons:
         apt:
           packages:
-          - libatlas-dev
-          - libatlas-base-dev
-          - liblapack-dev
-          - gfortran
-    - python: 2.7
-      env:
-        - TEST_AUTOWRAP="true"
-      sudo: true
+            - gfortran
+            - gcc
+            - python-numpy
+            - cython
       virtualenv:
         system_site_packages: true
-    - python: 2.7
+    - python: 3.5 # Dummy value
       env:
-        - TEST_ASCII="true"
-    - python: 3.4
+        - TEST_SAGE="true"
+        - FASTCACHE="false"
+      sudo: true
+      dist: trusty
+
+    # These builds are allowed failures. They need to be duplicated here and below.
+    - python: "pypy"
       env:
-        - TEST_ASCII="true"
+        - TEST_DOCTESTS="true"
+        - FASTCACHE="false"
+    - python: "pypy"
+      env:
+        - TEST_SYMPY="true"
+        - SPLIT="1/4"
+        - FASTCACHE="false"
+    - python: "pypy"
+      env:
+        - TEST_SYMPY="true"
+        - SPLIT="2/4"
+        - FASTCACHE="false"
+    - python: "pypy"
+      env:
+        - TEST_SYMPY="true"
+        - SPLIT="3/4"
+        - FASTCACHE="false"
+    - python: "pypy"
+      env:
+        - TEST_SYMPY="true"
+        - SPLIT="4/4"
+        - FASTCACHE="false"
+
   allow_failures:
-    - python: 3.3 # Dummy value
+    - python: "pypy"
       env:
-        - TEST_SAGE="true"
+        - TEST_DOCTESTS="true"
+        - FASTCACHE="false"
+    - python: "pypy"
+      env:
+        - TEST_SYMPY="true"
+        - SPLIT="1/4"
+        - FASTCACHE="false"
+    - python: "pypy"
+      env:
+        - TEST_SYMPY="true"
+        - SPLIT="2/4"
+        - FASTCACHE="false"
+    - python: "pypy"
+      env:
+        - TEST_SYMPY="true"
+        - SPLIT="3/4"
         - FASTCACHE="false"
-      sudo: true
     - python: "pypy"
       env:
+        - TEST_SYMPY="true"
         - SPLIT="4/4"
+        - FASTCACHE="false"
+
 before_install:
   - if [[ "${FASTCACHE}" != "false" ]]; then
       pip install fastcache;
     fi
   - if [[ "${TEST_AUTOWRAP}" == "true" ]]; then
       pip uninstall -y numpy;
-      sudo apt-get update;
-      sudo apt-get install gfortran gcc python-numpy cython;
```

Comment:
[Here](https://github.com/sympy/sympy/pull/10084/files#diff-354f30a63fb0907d4ad57269548329e3R111). .travis.yml has specific syntax for installing apt packages.